packages/product-components/src/components/PassphraseTypeCard/useNonAsciiChars.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from 'react';
+
+// Regular expression to match non-ASCII characters
+const nonAsciiPattern = /[^\x00-\x7F]/g;
```

Comment:
The `nonAsciiPattern` (`/[^\x00-\x7F]/g`) together with `match` could make a nice utility in `@trezor/utils` package. Then, you’d only need to test that utility, avoiding the need to test the hook itself.

By the way, is the hook even necessary? It seems like it might be overkill.